changelog/0037-upgrade-to-elasticsearch7.yml

```diff
@@ -0,0 +1,350 @@
+  CaseES().aggregation(DateHistogram('in', 'inserted_at', 'month')).run()
+  FormES().aggregation(DateHistogram('in', 'inserted_at', 'month')).run()
+  ```
+
+  ##### Note 2: Resetting ES7 indices after dry-run
+
+  - To reset all ES7 indices, run below command from the release pointing to ES7
+    ```
+    ./manage.py initialize_es_indices --reset
+    ```
+  - You can specify an index via `--index` option to reset a single index.
+  - Note that this clears all the data
+
+  ##### Note 3: Additional setup steps to run ES2 and ES7 on same machines
+
+  Commcare-cloud doesn't support running ES2 and ES7 on same hosts. Some manual steps are required to run ES2 and ES7 run on same machines.
+
+  - Make sure that you have at least 60% free memory and at least 60% free storage 
+  - Update below settings in public.yml in the git es7-upgrade-branch 
+      ```
+      elasticsearch_tcp_port: must be other than 9300 or your corresponding ES2 tcp port
+      elasticsearch_http_port: must be other than 9200 or your corresponding ES2 http port
+      elasticsearch_memory: available free memory on your machines min 4GB
+      ```
+  - Create an elasticsearch7 systemd/upstart service manually.
+    - Stop the application with `cchq <env> downtime start`
+    - Stop Elasticsearch `cchq <env> service elasticsearch stop`
+    - Make a backup of elasticsearch 2 upstart conf file `cp /etc/init/elasticsearch.conf /etc/init/elasticsearch2.backup`  or systemd file `cp /etc/systemd/system/elasticsearch.service /etc/systemd/system/elasticsearch2.backup`
+    - Deploy ES7 with `cchq <env> aps --limit=elasticsearch --branch=<git-upgrade-branch>` This will delete ES2 systemd/upstart service but keep ES2 as is
+      - (If you are using upstart) Create elasticsearch7 service `cp /etc/init/elasticsearch.conf /etc/init/elasticsearch7.conf`
+      - (If you are using Systemd)  Create elasticsearch7 service `cp /etc/systemd/system/elasticsearch.service /etc/systemd/system/elasticsearch7.service` and `systemctl daemon-reload"`
+    - Restore ES2 service file 
+      - `cp /etc/systemd/system/elasticsearch2.backup /etc/systemd/system/elasticsearch.service`  and `systemctl daemon-reload"` 
+      - Or for upstart `cp /etc/init/elasticsearch2.backup /etc/init/elasticsearch.conf`
+    - Start ES2 by `sudo service elasticsearch start`
+    - Start ES7 by `sudo service elasticsearch7 start`
+
+  You may now proceed from rest of the steps from Setup phase
+
+
+  ##### Note 4 - Upgrade without extra hardware resources
+
+  If you do not have access to any spare resources and would like to simply replace Elasticsearch 2 with 7, you can delete the existing Elasticsearch 2 cluster and deploy Elasticsearch 7 cluster. This will require downtime while the data is being indexed into Elasticsearch 7. We strongly recommend against this approach as it could be slower to reindex data from HQ to Elasticsearch 7 during which your application needs to be down. Below are the steps to do this.
```

Comment:
```suggestion
  If you do not have access to any spare resources and would like to simply replace Elasticsearch 2 with 7, you can
  delete the existing Elasticsearch 2 cluster and deploy Elasticsearch 7 cluster. This will require downtime while the
  data is being indexed into Elasticsearch 7. We strongly recommend against this approach as it could be slower to
  reindex data from HQ to Elasticsearch 7 during which your application needs to be down. Below are the steps to do this.
```

Reply:
This makes the final output to be a multiline paragraph instead of single one

Reply:
Similar with other multiline suggestions.